testing/integration/kubernetes_agent_standalone_test.go

```diff
@@ -77,64 +77,87 @@ func TestKubernetesAgentStandaloneKustomize(t *testing.T) {
 		Group: define.Kubernetes,
 	})
 
+	ctx := context.Background()
 	kCtx := k8sGetContext(t, info)
 
-	renderedManifest, err := renderKustomize(agentK8SKustomize)
-	require.NoError(t, err, "failed to render kustomize")
+	nodeList := corev1.NodeList{}
+	err := kCtx.client.Resources().List(ctx, &nodeList)
+	require.NoError(t, err)
+
+	totalK8SNodes := len(nodeList.Items)
```

Comment:
Take care of filtering nodes here: we need to take out the tainted nodes with a `NoSchedule` effect  if agent does not have a toleration for that (ex. https://kubernetes.io/docs/reference/labels-annotations-taints/#node-role-kubernetes-io-control-plane-taint for control-plane nodes)

Reply:
more than a valid point! thanks for catching that @pchila 🙂